astropy/io/fits/column.py

```diff
@@ -1119,7 +1119,11 @@ def _verify_keywords(cls, name=None, format=None, unit=None, null=None,
                     "of formatting this column.")
 
             if dims_tuple:
-                if reduce(operator.mul, dims_tuple) > format.repeat:
+                if isinstance(recformat, _FormatP):
+                    repeat = int(recformat.max)
+                else:
+                    repeat = format.repeat
+                if reduce(operator.mul, dims_tuple) > repeat:
```

Comment:
So this is to fix the warning with the second file from #7810 right ? (`S60mm-ICPBE-FEBEPAR.fits`)

Reply:
It seems TDIM is not taken into account though ?
```
In [7]: hdulist = fits.open('S60mm-ICPBE-FEBEPAR.fits')
   ...: data = hdulist[1].data

In [8]: data['USEFEED']
Out[8]: _VLF([array([1, 1, 1, 1, 2, 2, 2, 2], dtype=int32)], dtype=object)

In [9]: data.columns['USEFEED']
Out[9]: name = 'USEFEED'; format = '1PJ(8)'; dim = '(8,1)'
```
(this is with your branch)

Reply:
> So this is to fix the warning with the second file from #7810 right ? (`S60mm-ICPBE-FEBEPAR.fits`)

Correct.

Reply:
> It seems TDIM is not taken into account though ?
> ...
> (this is with your branch)

I'm having different results.... but just to be sure I have added a test about TDIM.